src/function/scalar/map/map.cpp

```diff
@@ -2,11 +2,56 @@
 #include "duckdb/common/string_util.hpp"
 #include "duckdb/parser/expression/bound_expression.hpp"
 #include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/function/aggregate/nested_functions.hpp"
 #include "duckdb/common/types/data_chunk.hpp"
 #include "duckdb/common/pair.hpp"
+#include "duckdb/common/types/value_map.hpp"
 
 namespace duckdb {
 
+void VerifyMap(Vector &map, idx_t count) {
+	D_ASSERT(map.GetType().id() == LogicalTypeId::MAP);
+	VectorData vdata;
+	map.Orrify(count, vdata);
+	auto map_validity = vdata.validity;
+
+	auto &children = StructVector::GetEntries(map);
+	VectorData key_vdata;
+	children[0]->Orrify(count, key_vdata);
+	auto key_data = (list_entry_t *)key_vdata.data;
+	auto key_validity = key_vdata.validity;
+
+	auto &key_entries = ListVector::GetEntry(*children[0]);
+	VectorData key_entry_vdata;
+	key_entries.Orrify(count, key_entry_vdata);
+	auto entry_validity = key_entry_vdata.validity;
```

Comment:
`auto &`

Reply:
Ah I thought validity was a pointer like data for some reason 😅